src/app/dspace/authorization/login/login-modal.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component } from 'angular2/core';
+import { Router } from 'angular2/router';
+
+import { FormBuilder, NgForm } from 'angular2/common';
+
+import { TranslateService, TranslatePipe } from "ng2-translate/ng2-translate";
+
+import { AuthorizationService } from '../services/authorization.service';
+import { FormService } from '../../../utilities/form/form.service';
+
+import { FormFieldsetComponent } from '../../../utilities/form/form-fieldset.component';
+import { FormModalComponent, ModalAction } from '../../../utilities/form/form-modal.component';
+import { LoginComponent } from './login.component';
+
+/**
+ * Login form. Uses form-modal component.
+ */
+@Component({
+  	selector: 'login-modal',
+  	directives: [ FormFieldsetComponent, FormModalComponent ],
+  	pipes: [ TranslatePipe ],
+  	template: `
+                <form-modal *ngIf="active" 
+                    [form]="form"
+                    [title]="'login.title'"
+                    [cancelLabel]="'login.cancel'"
+                    [confirmLabel]="'login.confirm'"
+                    [disabled]="disabled()"
+                    (loadedEmitter)="onLoaded($event)"
+                    (actionEmitter)="onAction($event)">
+                    <form-fieldset [form]="form" [inputs]="inputs"></form-fieldset>
+                    <span *ngIf="unauthorized" class="validaiton-helper text-danger">
+                        {{ 'login.unauthorized' | translate }}
+                    </span>
+                </form-modal>
+              `
+})
+export class LoginModalComponent extends LoginComponent {
+
+    /**
+     * Actual FormModal used to show and hide modal.
+     */
+    private login: FormModalComponent;
+
+    /**
+     *
+     * @param translate
+     *      TranslateService
+     * @param formService
+     *      FormService is a singleton service to retrieve form data.
+     * @param builder
+     *      FormBuilder is a singleton service provided by Angular2.
+     * @param authorization
+     *      AuthorizationService is a singleton service to interact with the authorization service.
+     * @param router
+     *      Router is a singleton service provided by Angular2.
+     */
+    constructor(private translate: TranslateService,
+                formService: FormService,
+                builder: FormBuilder,
+                authorization: AuthorizationService,
+                router: Router) {
+        super(formService, builder, authorization, router);
+        translate.setDefaultLang('en');
```

Comment:
I have been removing these around the code when I encountered them. I initially did the translation in this way, but the translate.setdefault, translate.use should only be included in app.component.ts.

The reason for this: The TranslateService is a singleton, if we just set one time, those values will be saved. The exception is if you include the TranslateService as a 'providers' property, then a new instance of the service will be created without the language to use or the default language set.